### DISCUSSION TITLE
Decimal Type converter error

### DISCUSSION THREAD

**Discussion by rakeshmurthy** [JoshClose/CsvHelper#1729] [Q&A]
```
I have mapped using index
```cs
 Map(m => m.Rate).Index(12).TypeConverterOption.NumberStyles(NumberStyles.Number);
 Map(m => m.Quantity).Index(13);
 Map(m => m.Amount).Index(14).TypeConverterOption.NumberStyles(NumberStyles.Number);

var csvConfig = new CsvConfiguration(CultureInfo.CurrentCulture)
                {
                    HasHeaderRecord = false,
                    Delimiter = ",",
                    MissingFieldFound = null,
                };
                using (var reader = new StringReader(line))
                using (var csv = new CsvReader(reader,csvConfig))
                {
                    csv.Context.RegisterClassMap<VendorMap>();
                    vendorList = csv.GetRecords<Vendor>().ToList();
                }
```
I still get Decimal converter error. I am using 25.0.0

I have checked all your answers related to this. For example this

https://github.com/JoshClose/CsvHelper/issues/1125#issuecomment-424761896

**Comment by JoshClose** [comment]
Is there a way to `decimal.Parse("**29.45**")`?

**Comment by rakeshmurthy** [comment]
Sorry I did not get you? Where shall this be applied? in Mapper ?

**Comment by JoshClose** [comment]
I'm saying, is there a way outside of CsvHelper to convert the string `**29.45**` to a `decimal`?

**Comment by JoshClose** [comment]
Because `**29.45**` isn't a number. I'm just checking to make sure before I suggest how to remove the `*`'s

**Comment by rakeshmurthy** [comment]
Sorry its was supposed to be shown as **29.45** ..

The actual text is 29.45. No **

**Comment by JoshClose** [comment]
Oh, lol. What is the value of `CurrentCulture`?

**Comment by rakeshmurthy** [comment]
CultureInfo.CurrentCulture = en-US
```cs
 var csvConfig = new CsvConfiguration(CultureInfo.CurrentCulture)
                {
                    HasHeaderRecord = false,
                    Delimiter = ",",
                    MissingFieldFound = null,
                };
                using (var reader = new StringReader(line))
                using (var csv = new CsvReader(reader,csvConfig))
                {
                    while (csv.Read())
                    {
                        csv.Context.RegisterClassMap<VendorMap>();
                        vendorList.Add(csv.GetRecord<Vendor>());
                    }
                }
```

**Comment by JoshClose** [comment]
There must be something else I'm missing. This works:
```cs
void Main()
{
	var s = "FLORENS,ML-CROWL-01,RNT1111585,05/31/2020,05/01/2020,05/31/2020,DFSU4055875,05/01/2020,05/31/2020,07/31/2019,,101,0.95,31,29.45";
	var config = new CsvConfiguration(CultureInfo.GetCultureInfo("en-US"))
	{
		HasHeaderRecord = false
	};
	using (var reader = new StringReader(s))
	using (var csv = new CsvReader(reader, config))
	{
		csv.Context.RegisterClassMap<FooMap>();
		csv.GetRecords<Foo>().ToList().Dump();
	}
}

private class Foo
{
	public decimal Decimal { get; set; }
}

private class FooMap : ClassMap<Foo>
{
	public FooMap()
	{
		Map(m => m.Decimal).Index(14).TypeConverterOption.NumberStyles(NumberStyles.Number);
	}
}
```
![image](https://user-images.githubusercontent.com/114971/109341844-98aa6800-7830-11eb-8ce7-cb0e140a9fad.png)

**Comment by rakeshmurthy** [comment]
I really donno what is missing. I just copy pasted your code. 

![image](https://user-images.githubusercontent.com/6312201/109342313-f5774400-786b-11eb-9f6c-dc366c4d98ac.png)

**Comment by JoshClose** [comment]
That is showing the text is empty. It looks like index 10 is the only empty one. What is the mapping for index 10?

**Comment by rakeshmurthy** [comment]
Map(m => m.EquipmentDropoffDate).Index(10); 

EquipmentDropoffDate is of datetime?

In the input this index is empty but i have made this nullable is it still a problem ? 

Actually ... there was problem with datetimeconverter as well. After I made all the datetime properties to nullable, then, that error did not appear

**Comment by JoshClose** [comment]
Everything good now?

**Comment by rakeshmurthy** [comment]
Unfortunately No...  can you please also include index 3,4,10,14 and try on your side?

**Comment by JoshClose** [comment]
Can you give me your class and mapping definitions so I can try the full thing?

**Comment by rakeshmurthy** [comment]
Ok I think i got the problem.

If i pass the data as string created like you have done, then, everything will work fine. 

However, if I read it from file and pass for some reason string value is going as "a,b,c,d"  with quotes... your way is without quotes..

**Comment by rakeshmurthy** [comment]
Thank you for your help. I solved it.. 

Can you please delete your comment https://github.com/JoshClose/CsvHelper/issues/1728#issuecomment-786838087 ?

Actually it is a actual class i am using. Please delete it from public forum.

**Comment by JoshClose** [comment]
Posts with the class in them deleted. What was the fix?

**Comment by rakeshmurthy** [comment]
Thank you so much. 

my string value was like following: 

var s = "\"a,b,c,d,e,f\"";

where as, string you created was 

var j = "a,b,c,d,e,f";

I removed the quotes from string s ... That solved the problem. 

Thank you so much for your quick help and response. Your CsvHelper is awesome and helping me a lot.

**Comment by JoshClose** [comment]
Great.
